pgcli/pgcompleter.py

```diff
@@ -384,6 +384,43 @@ def get_column_matches(self, suggestion, word_before_cursor):
 
         return self.find_matches(word_before_cursor, flat_cols, meta='column')
 
+    def get_join_matches(self, suggestion, word_before_cursor):
+        scoped_cols = self.populate_scoped_cols(suggestion.tables)
+        # Set up some data structures for efficient access
+        qualified = dict((t.ref, t.schema) for t in suggestion.tables)
+        tbls = set((t.schema, t.name) for t in scoped_cols.keys())
+        tblprio = dict((t.ref, n) for n, t in enumerate(suggestion.tables))
+        joins, prios = [], []
+        # Iterate over FKs in existing tables to find potential joins
+        for fk, rtbl, rcol in ((fk, rtbl, rcol)
+          for rtbl, rcols in scoped_cols.items()
+          for rcol in rcols
+          for fk in rcol.foreignkeys):
```

Comment:
This is really weird looking. I thought at first it was a mis-indented for-loop. Can you either switch to a traditional nested for-loop structure, or assign the generator expression to a named variable & loop over that variable?